fe/src/main/java/org/apache/doris/analysis/BinaryPredicate.java

```diff
@@ -225,7 +226,7 @@ public boolean equals(Object obj) {
         if (!super.equals(obj)) {
             return false;
         }
-        return ((BinaryPredicate) obj).opcode == this.opcode;
+        return ((BinaryPredicate) obj).opcode == this.opcode && op.equals(((BinaryPredicate) obj).getOp());
```

Comment:
I think only compare op is enough.

Reply:
It only for safe, don't know what the `opcode` for.
If it is useless, we can remove it in next PR.